src/test/java/me/karakelley/http/controllers/StaticFilesControllerTest.java

```diff
@@ -22,39 +21,65 @@
-      assertTrue(response.getBody().split("", 2).length == 2);
+      assertTrue(new String(response.getBody()).split("", 2).length == 2);
     });
   }
 
   @Test
   void testPostDisplayFile() {
-    ContentGenerator contentGenerator = new HtmlGenerator();
     PublicDirectory publicDirectory = PublicDirectory.create("/", new FileFinderCache(new RealFileFinder()));
-    Controller controller = new StaticFilesController(publicDirectory, contentGenerator);
+    Controller controller = new StaticFilesController(publicDirectory, new HtmlGenerator());
     Response response = controller.respond(new Request(newBufferedReader("POST / HTTP/1.1\r\n"), 0));
-    assertEquals(response.deliver(), "HTTP/1.1 404 Not Found\r\n\r\n");
+    assertEquals(new String(response.deliver()), "HTTP/1.1 404 Not Found\r\n\r\n");
   }
 
   @Test
   void testCreatesLinksForDirectories() {
-    PublicDirectory publicDirectory = PublicDirectory.create("./src/", new FileFinderCache(new RealFileFinder()));
+    TempFilesHelper.withTempDirectory(directory -> {
+      Path file = TempFilesHelper.createTempFile(directory, "/test1");
+      PublicDirectory publicDirectory = PublicDirectory.create(directory.toString(), new FileFinderCache(new RealFileFinder()));
+      ContentGenerator contentGenerator = new HtmlGenerator();
+      Controller controller = new StaticFilesController(publicDirectory, contentGenerator);
+      Response response = controller.respond(new Request(newBufferedReader("GET / HTTP/1.1\r\n"), 0));
 
-    ContentGenerator contentGenerator = new HtmlGenerator();
-    Controller controller = new StaticFilesController(publicDirectory, contentGenerator);
-    Response response = controller.respond(new Request(newBufferedReader("GET /test HTTP/1.1\r\n"), 0));
-    assertTrue(response.deliver().contains("<p><a href="));
+      assertTrue(new String(response.getBody()).contains("<p><a href=\"/test1.txt\">test1.txt</a></p>"));
+    });
+  }
+
+  @Test
+  void testServesTextFile() {
+    TempFilesHelper.withTempDirectory(directory -> {
+      Path file = TempFilesHelper.createTempFile(directory, "/test1");
+      writeToFile("Hello World", file);
+
+      PublicDirectory publicDirectory = PublicDirectory.create(directory.toString(), new FileFinderCache(new RealFileFinder()));
+      ContentGenerator contentGenerator = new HtmlGenerator();
+      Controller controller = new StaticFilesController(publicDirectory, contentGenerator);
+      Response response = controller.respond(new Request(newBufferedReader("GET /test1.txt HTTP/1.1\r\n"), 0));
+      assertEquals("Hello World", new String(response.getBody()));
+    });
   }
 
   private LineReader newBufferedReader(String request) {
     ByteArrayInputStream inputStream = new ByteArrayInputStream(request.getBytes());
     return new BufferedLineReader(new InputStreamReader(inputStream));
   }
+
+  private void writeToFile(String text, Path file) {
```

Comment:
This is in two different test files, and probably belongs in neither. There are a couple different styles to choose from that would be a little more ergonomic. One suggestion is something like:
```
TempFilesHelper.createTempFile(directory, "/test1").withContents("Hello World")
```